src/paths/shortest_paths.c

```diff
@@ -1275,3 +1276,326 @@ int igraph_diameter_dijkstra(const igraph_t *graph,
+ *           Invalid argument.
+ *        \endclist
+ *
+ * Time complexity: K|V|(|V|log|V| + |E|), where |V| is the number of vertices,
+ *                  and |E| is the number of edges.
+ */
+int igraph_k_shortest_paths(const igraph_t *graph,
+                            igraph_vector_t *weights,
+                            igraph_vector_ptr_t *paths,
+                            igraph_integer_t k,
+                            igraph_integer_t from,
+                            igraph_integer_t to,
+                            igraph_neimode_t mode) {
+    igraph_vector_ptr_t paths_pot; /* potential shortest paths */
+    igraph_integer_t vertex_spur;
+    igraph_vector_t path_spur;
+    igraph_vector_t *path_root, *path_total;
+    igraph_integer_t nr_edges_root, i_path_current, i_path, edge_path_root, vertex_root_del;
+    long int i;
+    igraph_vector_t weights_old;
+    igraph_vector_t edges_removed;
+    igraph_bool_t added_path, null_weights;
+    long int nr_edges = igraph_ecount(graph);
+    igraph_bool_t infinite_path, already_in_potential_paths;
+    igraph_vector_t *path_0;
+
+    igraph_vector_ptr_free_all(paths);
+    IGRAPH_VECTOR_PTR_SET_ITEM_DESTRUCTOR(paths, igraph_vector_destroy);
+    igraph_vector_ptr_resize(paths, k);
+
+    if (k == 0) {
+        return IGRAPH_SUCCESS;
+    }
+
+    for (i = 1; i < k; i ++) {
+        VECTOR(*paths)[i] = NULL;
+    }
+
+    VECTOR(*paths)[0] = IGRAPH_CALLOC(1, igraph_vector_t);
+    path_0 = VECTOR(*paths)[0];
+    igraph_vector_init(path_0, 0);
+
+    IGRAPH_CHECK(igraph_get_shortest_path_dijkstra(graph,
+                                                   NULL,
+                                                   path_0,
+                                                   from,
+                                                   to,
+                                                   weights,
+                                                   mode));
+
+    /* Check if there's a path. */
+    infinite_path = 0;
+    for (i = 0; i < igraph_vector_size(path_0); i++) {
+        int edge = VECTOR(*path_0)[i];
+        if (weights && !IGRAPH_FINITE(VECTOR(*weights)[edge])) {
+            infinite_path = 1;
+        }
+    }
+    if (infinite_path || (from != to && igraph_vector_size(path_0) == 0)) {
+        /* No path found. */
+        igraph_vector_destroy(path_0);
+        IGRAPH_FREE(path_0);
+        igraph_vector_ptr_resize(paths, 0);
+        return IGRAPH_SUCCESS;
+    }
+
+    igraph_vector_ptr_init(&paths_pot, 0);
+    IGRAPH_VECTOR_PTR_SET_ITEM_DESTRUCTOR(&paths_pot, igraph_vector_destroy);
+    IGRAPH_FINALLY(igraph_vector_ptr_destroy_all, &paths_pot);
+    IGRAPH_VECTOR_INIT_FINALLY(&path_spur, 0);
+    IGRAPH_VECTOR_INIT_FINALLY(&weights_old, 0);
+    IGRAPH_VECTOR_INIT_FINALLY(&edges_removed, 0);
+
+    /* If weights are NULL we use weights of 1, which then later can be replaced
+     * with infinite weights.
+     */
+    null_weights = !weights;
```

Comment:
`no_weights` to avoid confusion with 0 values.

Reply:
Renamed to `weights_missing` as `no_weights` can be mistaken for "number of weights".